### PR TITLE
Make the SVGs responsive

### LINKED ISSUE #173: screen resize issue
There's a small transition period with snafu (e.g. at ~1 seconds and ~11 seconds)

https://www.youtube.com/watch?v=EJxfIclhxVA

### PR DESCRIPTION
**Pull request by IsmailM**
This makes the SVGs responsive (and as such it fixes #173), by redrawing the SVG on a browser resize...
<br>

NOTE: The first commit is an alternative solution, which calculated an aspect ratio to resize the SVG... However, I found that the text within the SVG was also resized. As a result, the SVG axis font would sometimes be too small (when decreasing browser width) or too big (when increasing browser width), which made the graphical overview chart look really weird... Thus I came up with this alternative solution...

What do you think?

### CODE REVIEW

**Comment by yeban** [comment]
I can took a look into this Wednesday or later. Perhaps @vivekiitkgp will be able to advise in the mean time. He wrote the graphical overview.

File: public/js/jquery.t.js [hunk 1]
```diff
@@ -16,6 +16,23 @@
         });
     };
 
+    var debounce = function(fn, timeout) {
```
Comment (vivekiitkgp) [inline public/js/jquery.t.js]: This should be `function (fn, timeout)`

File: public/js/jquery.t.js [hunk 2]
```diff
@@ -16,6 +16,23 @@
...
+      var timeoutID = -1;
+      return function() {
```
Comment (vivekiitkgp) [inline public/js/jquery.t.js]: Same here.

File: public/js/jquery.t.js [hunk 3]
```diff
@@ -16,6 +16,23 @@
...
+        if (timeoutID > -1) {
+          window.clearTimeout(timeoutID);
+        }
+        timeoutID = window.setTimeout(fn, timeout);
+      }
+    };
+   
+    var setupResponsiveness = function ($queryDiv, $graphDiv, index, howMany, opts)  {
+        var debounced_draw = debounce(function() {
```
Comment (vivekiitkgp) [inline public/js/jquery.t.js]: Same here.

File: public/js/jquery.t.js [hunk 4]
```diff
@@ -348,6 +365,8 @@
             // Ensure clicking on 'rect' takes user to the relevant hit on all
             // browsers.
             setupClick($graphDiv);
+            // Redraw the graph on a browser resize...
+            setupResponsiveness($queryDiv, $graphDiv, index, howMany, opts, resize);
```
Comment (vivekiitkgp) [inline public/js/jquery.t.js]: `setupResponsiveness` doesn't take the `resize` parameter.

File: public/js/jquery.t.js [hunk 5]
```diff
@@ -16,6 +16,23 @@
...
     };
 
+    var debounce = function(fn, timeout) {
+      var timeoutID = -1;
+      return function() {
+        if (timeoutID > -1) {
+          window.clearTimeout(timeoutID);
+        }
+        timeoutID = window.setTimeout(fn, timeout);
+      }
```
Comment (vivekiitkgp) [inline public/js/jquery.t.js]: Semicolon

File: public/js/jquery.t.js [hunk 6]
```diff
@@ -340,7 +357,7 @@
             }
             // Bind listener events once all the graphical elements have
             // been drawn for first time.
-            if (index === 0) {
+            if (index === 0 && resize !== 1 ) {
```
Comment (vivekiitkgp) [inline public/js/jquery.t.js]: Can you document the `resize` parameter (as in, why it is important)? May be in `setupResponsiveness` or here itself.

**Comment by vivekiitkgp** [comment]
@IsmailM I have added few comments. Once they are addressed, it is good to go.

File: public/js/jquery.t.js [hunk 7]
```diff
@@ -16,6 +16,23 @@
...
+    var debounce = function(fn, timeout) {
+      var timeoutID = -1;
+      return function() {
+        if (timeoutID > -1) {
+          window.clearTimeout(timeoutID);
+        }
+        timeoutID = window.setTimeout(fn, timeout);
+      }
+    };
+
```
Comment (vivekiitkgp) [inline public/js/jquery.t.js]: Trailing spaces here.

**Comment by IsmailM** [comment]
@vivekiitkgp 
Thanks for having a look... I've tidied it up...

**Comment by vivekiitkgp** [comment]
@IsmailM Probably squash the last commit into the previous one? I feel that having a separate commit to fix minor things would add to the log noise, given that you can integrate those changes in the main commits.

It's not an issue per se, just a personal thought ;) Rest is cool.

**Pull request closed without merging**